### PR TITLE
Compound color pass

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/internal-planning/issues/450
Fixes https://github.com/vector-im/element-web/issues/25547

------

Refer to the [[Compound token semantics documentation](http://localhost:8080/#/room/!aYfWTUNfRxRtgOtTcy:matrix.org)]. 

**Few note-worthy thing for review context**

- All SCSS `rgba($..., alpha)` logic does not work with Compound as CSS logical properties can have a different values based on where they exist within the DOM.
- I've introduced a `$accent-XXX` semantic scale, this exists within Element and not Compound as the new visual language does not need this semantic. Should pretty much never be used, but is helpful for the transition period
- A few deprecated things have been removed, but at large this is a simple replacement of theme values 
- All the icons that are currently imported is because we need to tweak the webpack config for it to look at the right place and this comes in the way of testing the prototype early.

<h2>Preview</h2>

Screenshot 2023-07-05 at 15 51 04

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Compound color pass ([\#11079](https://github.com/matrix-org/matrix-react-sdk/pull/11079)). Fixes vector-im/internal-planning#450 and vector-im/element-web#25547.

### CODE REVIEW

File: res/themes/light/icons/$icons.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"icon":{"chat":{"value":"icons/chat.svg","type":"icon"},"check-circle":{"value":"icons/check-circle.svg","type":"icon"},"check":{"value":"icons/check.svg","type":"icon"},"chevron":{"value":"icons/chevron.svg","type":"icon"},"close":{"value":"icons/close.svg","type":"icon"},"computer":{"value":"icons/computer.svg","type":"icon"},"delete":{"value":"icons/delete.svg","type":"icon"},"error":{"value":"icons/error.svg","type":"icon"},"info":{"value":"icons/info.svg","type":"icon"},"lock":{"value":"icons/lock.svg","type":"icon"},"mobile":{"value":"icons/mobile.svg","type":"icon"},"thread":{"value":"icons/thread.svg","type":"icon"},"user":{"value":"icons/user.svg","type":"icon"},"visibility-invisible":{"value":"icons/visibility-invisible.svg","type":"icon"},"visibility-visible":{"value":"icons/visibility-visible.svg","type":"icon"},"web-browser":{"value":"icons/web-browser.svg","type":"icon"}}}
```
Comment (t3chguy) [inline res/themes/light/icons/$icons.json:1]: What is this file for? What generates it? What reads it?

Comment (gsouquet) [reply res/themes/light/icons/$icons.json:1]: This should not be included, I've explained it in this pull request description.

Comment (t3chguy) [reply res/themes/light/icons/$icons.json:1]: Sorry I must be blind, I see explanation for why all the SVGs are included, but nothing about this file?

Comment (gsouquet) [reply res/themes/light/icons/$icons.json:1]: It's just a copy paste of the output of what is in the design tokens. It won't be there when released

Comment (t3chguy) [reply res/themes/light/icons/$icons.json:1]: What reads this file on this side of things?

File: res/css/structures/_LeftPanel.pcss [hunk 2]
```diff
@@ -14,13 +14,6 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-.mx_MatrixChat--with-avatar {
-    .mx_LeftPanel,
-    .mx_LeftPanel .mx_LeftPanel_roomListContainer {
-        background-color: transparent;
-    }
-}
```
Comment (t3chguy) [inline res/css/structures/_LeftPanel.pcss:22]: What's the reason for this being removed?

File: res/css/views/rooms/_EditMessageComposer.pcss [hunk 3]
```diff
@@ -33,7 +33,7 @@ limitations under the License.
         padding: 3px 6px;
 
         &:focus {
-            border-color: rgba($accent, 0.5); /* Only ever used here */
+            border-color: $accent-500;
```
Comment (t3chguy) [inline res/css/views/rooms/_EditMessageComposer.pcss:36]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_MessageComposer.pcss [hunk 4]
```diff
@@ -197,11 +197,12 @@ limitations under the License.
 
     &.mx_MessageComposer_closeButtonMenu {
         &::after {
-            background: rgba($accent, 0.1);
+            background: $accent-300;
```
Comment (t3chguy) [inline res/css/views/rooms/_MessageComposer.pcss:200]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_MessageComposer.pcss [hunk 5]
```diff
@@ -235,11 +236,12 @@ limitations under the License.
 
         &.mx_MessageComposer_closeButtonMenu {
             &::after {
-                background: rgba($accent, 0.1);
+                background: $accent-300;
```
Comment (t3chguy) [inline res/css/views/rooms/_MessageComposer.pcss:239]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_RecentlyViewedButton.pcss [hunk 6]
```diff
@@ -19,7 +19,7 @@ limitations under the License.
     width: max-content;
     max-width: 240px;
     max-height: 400px;
-    border: 1px solid rgba($primary-content, 0.1);
+    border: 1px solid var(--cpd-color-gray-300);
```
Comment (t3chguy) [inline res/css/views/rooms/_RecentlyViewedButton.pcss:22]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_RoomHeader.pcss [hunk 7]
```diff
@@ -56,7 +56,7 @@ limitations under the License.
     min-width: 0;
     margin: 0 20px 0 16px;
     padding-top: 6px;
-    border-bottom: 1px solid $system;
+    border-bottom: 1px solid var(--cpd-color-gray-500);
```
Comment (t3chguy) [inline res/css/views/rooms/_RoomHeader.pcss:59]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_RoomHeader.pcss [hunk 8]
```diff
@@ -112,7 +112,7 @@ limitations under the License.
     }
 
     &[aria-expanded="true"] {
-        background-color: $quinary-content;
+        background-color: var(--cpd-color-gray-400);
```
Comment (t3chguy) [inline res/css/views/rooms/_RoomHeader.pcss:115]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/rooms/_RoomHeader.pcss [hunk 9]
```diff
@@ -191,7 +191,7 @@ limitations under the License.
     }
 
     &:hover {
-        background: rgba($accent, 0.1);
+        background: $accent-300;
```
Comment (t3chguy) [inline res/css/views/rooms/_RoomHeader.pcss:194]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/css/views/settings/_LayoutSwitcher.pcss [hunk 10]
```diff
@@ -28,6 +28,7 @@ limitations under the License.
         flex-shrink: 1;
         display: flex;
         flex-direction: column;
+        overflow: hidden;
```
Comment (t3chguy) [inline res/css/views/settings/_LayoutSwitcher.pcss:31]: Why this change?

File: res/css/views/settings/_LayoutSwitcher.pcss [hunk 11]
```diff
@@ -71,7 +72,7 @@ limitations under the License.
     }
 
     .mx_StyledRadioButton_checked {
-        background-color: rgba($accent, 0.08);
+        background-color: $accent-200;
```
Comment (t3chguy) [inline res/css/views/settings/_LayoutSwitcher.pcss:75]: This will break custom themes e.g. https://github.com/aaronraimist/element-themes

File: res/themes/legacy-light/css/_legacy-light.pcss [hunk 12]
```diff
@@ -194,6 +189,31 @@ $username-variant6-color: #2dc2c5;
 $username-variant7-color: #5c56f5;
 $username-variant8-color: #74d12c;
```
Comment (t3chguy) [inline res/themes/legacy-light/css/_legacy-light.pcss:190]: Should these username colour variants not be mapped using the colour scale?

**Comment by germain-gg** [comment]
@t3chguy I will mark all the "custom theme" comments as resolved as I believe this is the right approach for those.

What needs to happen is to expose the `accent-HUE` values in `_custom.pcss` for folks to override those values and them provide them with a tool to generate the color scale themselves.

**Comment by t3chguy** [comment]
> I believe this is the right approach for those.

Sorry please clarify what the right approach it? Shipping a build which breaks custom themes? Or fixing it in the manner you describe before the PR lands?

File: res/css/views/settings/_LayoutSwitcher.pcss (hunk 10 above)
Comment (germain-gg) [reply res/css/views/settings/_LayoutSwitcher.pcss:31]: There was an issue with the background bleeding over the border radius. And `overflow: hidden` fixed the issue in this context. Weirdly happened in only one browser

Comment (t3chguy) [reply res/css/views/settings/_LayoutSwitcher.pcss:31]: Is this specific to the colour pass?

**Comment by germain-gg** [comment]
I have not advocated to leave people with no solutions. We will expose all the variables from `$accent-100` to `$accent-1400` to `_custom.pcss` and provide anyone that wishes to, a script to generate a custom color scale given an accent color.

I do not believe we need to embed more complexity within Element Web by dynamically generating those color scale.

File: res/css/views/settings/_LayoutSwitcher.pcss (hunk 10 above)
Comment (germain-gg) [reply res/css/views/settings/_LayoutSwitcher.pcss:31]: Yes, i have not seen this issue occur in mainline. But i have not looked for the root cause.

**Review by janogarcia — CHANGES_REQUESTED** [review]
@germain-gg Heads up that **tooltips in dark theme** are not rendering correctly (black text on dark background).

Also, [suggested some tweaks on Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=0-1&mode=design&t=Cfyv4DhYivJXGyUU-0).

**Review by janogarcia — CHANGES_REQUESTED** [review]
@germain-gg Please find the review comments and requested changes on the annotated screenshots on [Figma: Review Round 3](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=96-79&mode=design&t=s6U1wFY3x86j6OA0-0). 👍

**Comment by germain-gg** [comment]
@janogarcia updated everything but the pills. Can you re-review?

**Comment by janogarcia** [comment]
@germain-gg I'm on it. 👍 What is stopping us from updating the pills? I just realized they have a hover state, but we could easily come up with one.

**Review by janogarcia — CHANGES_REQUESTED** [review]
@germain-gg Review comments and visual annotations available on [Figma: Review Round 4](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=104-173&mode=design&t=s6U1wFY3x86j6OA0-0).

File: res/css/views/rooms/_EventTile.pcss [hunk 13]
```diff
@@ -1136,7 +1136,7 @@ $left-gutter: 64px;
     }
 
     &:hover {
-        background-color: $system;
+        background-color: var(--cpd-color-bg-subtle-secondary);
```
Comment (t3chguy) [inline res/css/views/rooms/_EventTile.pcss:1139]: Looks like this will be no longer flexible for custom themes

File: res/css/views/rooms/_ThreadSummary.pcss [hunk 14]
```diff
@@ -55,8 +55,8 @@ limitations under the License.
         bottom: 0;
         width: 60px;
         box-sizing: border-box;
-        /* XXX: We use `$system-transparent` instead of `transparent` to work around a Safari <15.4 bug */
-        background: linear-gradient(270deg, $system 50%, $system-transparent 100%);
+        /* XXX: We use an HEXA `transparent` to work around a Safari <15.4 bug */
+        background: linear-gradient(270deg, $system 50%, #e1e6ec00 100%);
```
Comment (t3chguy) [inline res/css/views/rooms/_ThreadSummary.pcss:59]: Looks like this will be no longer flexible for custom themes

File: res/themes/light-custom/css/_custom.pcss [hunk 15]
```diff
@@ -26,16 +26,10 @@ $secondary-content: var(--secondary-content, $secondary-content);
...
 $system: var(--system, $system);
 $background: var(--background, $background);
-$panels: rgba($system, 0.9);
-$panel-base: var(--panel-base, $tertiary-content); /* This color is not intended for use in the app */
-$panel-selected: rgba($panel-base, 0.3);
-$panel-hover: rgba($panel-base, 0.1);
-$panel-actions: rgba($panel-base, 0.2);
-$space-nav: rgba($panel-base, 0.1);
+$panels: var(--cpd-color-gray-600);
+$panel-actions: var(--cpd-color-gray-300);
```
Comment (t3chguy) [inline res/themes/light-custom/css/_custom.pcss:32]: I don't think we should be leaking `--cpd-X` into custom theme var naming, this could be the default value of course but I wouldn't expect to need to use `"cpd-color-gray-600": foo` in the JSON.

File: res/themes/light-custom/css/_custom.pcss [hunk 16]
```diff
@@ -145,3 +139,23 @@ $menu-selected-color: var(--menu-selected-color, $menu-selected-color);
 $pill-bg-color: var(--other-user-pill-bg-color, $pill-bg-color);
 $pill-hover-bg-color: var(--other-user-pill-bg-color, $pill-hover-bg-color);
 $icon-button-color: var(--icon-button-color, $icon-button-color);
+
+/**
+ * Variation of the accent color.
+ * Generate this colour scale using LeonardoColor
+ * TODO: Add link
```
Comment (t3chguy) [inline res/themes/light-custom/css/_custom.pcss:146]: Pending TODO, is this going to be done as part of this PR?

File: res/css/_common.pcss [hunk 17]
```diff
@@ -263,7 +263,7 @@ legend {
     :not(.mx_textinput):not(.mx_Field):not(.mx_no_textinput) > input[type="text"]::placeholder,
     :not(.mx_textinput):not(.mx_Field):not(.mx_no_textinput) > input[type="search"]::placeholder,
     .mx_textinput input::placeholder {
-        color: rgba($input-darker-fg-color, 0.75);
+        color: var(--cpd-color-text-placeholder);
```
Comment (t3chguy) [inline res/css/_common.pcss:266]: This will no longer be flexible to custom themes

File: res/css/_common.pcss [hunk 18]
```diff
@@ -250,7 +250,7 @@ legend {
         background-color: transparent;
         color: $input-darker-fg-color;
         border-radius: 4px;
-        border: 1px solid rgba($primary-content, 0.1);
+        border: 1px solid var(--cpd-color-gray-300);
```
Comment (t3chguy) [inline res/css/_common.pcss:253]: This will no longer be flexible to custom themes

File: res/themes/light-custom/css/_custom.pcss (hunk 16 above)
Comment (germain-gg) [reply res/themes/light-custom/css/_custom.pcss:146]: No, it won't be

Comment (t3chguy) [reply res/themes/light-custom/css/_custom.pcss:146]: Then please open an issue to track the work and link it here :D

File: res/themes/light-custom/css/_custom.pcss (hunk 15 above)
Comment (germain-gg) [reply res/themes/light-custom/css/_custom.pcss:32]: Makes sense! 

changed to have `--panels` and use `--cpd-color-gray-X` as the default value

File: res/css/views/rooms/_ThreadSummary.pcss (hunk 14 above)
Comment (germain-gg) [reply res/css/views/rooms/_ThreadSummary.pcss:59]: I don't think i have a better solution for this `#e1e6ec00` this would require a color function in any case to be backwards compatible. 

This will do a transition to a hardcoded transparent hex code, which seems good enough given the name of the variable 🤔

Do you see alternatives here?

Comment (t3chguy) [reply res/css/views/rooms/_ThreadSummary.pcss:59]: Could we not keep the `$system-transparent` variable and set it to `#e1e6ec00` in the light base theme so all themes inherit, and let `_custom` override it with `var(--system-transparent)`

File: res/themes/light-custom/css/_custom.pcss (hunk 16 above)
Comment (germain-gg) [reply res/themes/light-custom/css/_custom.pcss:148]: https://github.com/vector-im/compound/issues/158

**Review by janogarcia — COMMENTED** [review]
Secondary text in timeline (disambiguated names, "edited" flag, room update): [Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=119-209&mode=design&t=k1essnYIgHXlWGEe-0)

The token seem to have been applied but there's still an opacity value being applied which makes basically breaks the expected rendered color.

---

Sessions: [Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=119-209&mode=design&t=k1essnYIgHXlWGEe-0)

It seems you didn't notice the screenshots annotations on the bottom.

---

Reaction: [Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=119-209&mode=design&t=k1essnYIgHXlWGEe-0)

I'm still seeing the previous color treatment. Nothing seems to have changed.

---

Composer overflow icons: [Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=119-209&mode=design&t=k1essnYIgHXlWGEe-0)

They are still using `color.text.secondary` instead of `color.icon.tertiary` 

---

Files panel: [Figma](https://www.figma.com/file/GmiEFTUaqAulfYzou1Rwrz/Review%3A-Germain's-Typography-and-Color-pass?type=design&node-id=119-209&mode=design&t=k1essnYIgHXlWGEe-0)

Could we slightly clean up / implement the suggested tweaks? Otherwise, the file panel looks even more broken than how it currently is.

**Review by janogarcia — APPROVED** [review]

**Comment by janogarcia** [comment]
Forgot to add that I approved the changes (before the deployment landed) based on the call where we went over the remaining details.

**Review by t3chguy — APPROVED** [review]
:shipit:
